packages/compass-components/src/components/document-list/element.tsx

```diff
@@ -420,6 +420,29 @@ const isValidUrl = (str: string): boolean => {
   }
 };
 
+/**
+ * Helper function to get the nested key path of an element, skips array keys
+ * Meant for keypaths used in query conditions from a selected element
+ */
+const getNestedKeyPath = (element: HadronElementType): string => {
```

Comment:
Not sure if this fully replicates what you meant with schema visualization but this approach seems to work, the only caveat here is that with arrays one can only add to query 1 nested item at a time.
Alternatively we can make it specify the index (which would enforce its position.... not sure how common that is) or manually handle it to be smarter about it.

IMO, as a first iteration one item a time isn't too bad. We can have multiple nested fields for the object in the meantime so that's good

Reply:
We already handle this in schema tab with `toggleDistinctValue` case (or as we only have "add" here, you can look at `addDistinctValue`), I think for an "add to query" option it makes more sense than overriding the existing value